In [1]:
!conda env list


# conda environments:
#
# * -> active
# + -> frozen
                         /home/njm12/ATMS_523/envs/era5-850
                         /home/njm12/ATMS_523/envs/xarray-climate
base                 *   /opt/conda



In [2]:
!conda run -p /home/njm12/ATMS_523/envs/xarray-climate python -m ipykernel install --user --name xarray-climate --display-name "Python (xarray-climate)"

Installed kernelspec xarray-climate in /home/njm12/.local/share/jupyter/kernels/xarray-climate



In [3]:
import sys
print(sys.executable)

/home/njm12/ATMS_523/envs/xarray-climate/bin/python


In [4]:
# ================================================================
# Median Path Length Table by Season and Group (CORRECTED)
# ================================================================

import pandas as pd

median_data = {
    "Season": ["DJF","DJF","DJF","MAM","MAM","MAM","JJA","JJA","JJA","SON","SON","SON"],
    "Group": ["Crossed River","East Side","West Side"]*4,
    "Median_km": [
        # DJF (corrected)
        36.210150, 5.246448, 7.467338,

        # MAM (corrected)
        13.116121, 5.230355, 4.345218,

        # JJA (corrected)
        7.579991, 1.786367, 2.574944,

        # SON (corrected)
        26.505830, 7.242030, 4.023350
    ]
}

median_df = pd.DataFrame(median_data)
median_df

,Season,Group,Median_km
0,DJF,Crossed River,36.210150
1,DJF,East Side,5.246448
2,DJF,West Side,7.467338
3,MAM,Crossed River,13.116121
4,MAM,East Side,5.230355
5,MAM,West Side,4.345218
6,JJA,Crossed River,7.579991
7,JJA,East Side,1.786367
8,JJA,West Side,2.574944
9,SON,Crossed River,26.505830


In [5]:
# ================================================================
# Compute Median Differences (Absolute + Percent)
# ================================================================

results = []

for season in median_df["Season"].unique():
    
    subset = median_df[median_df["Season"] == season].set_index("Group")
    
    cr = subset.loc["Crossed River","Median_km"]
    east = subset.loc["East Side","Median_km"]
    west = subset.loc["West Side","Median_km"]
    
    # Absolute differences
    diff_cr_east = cr - east
    diff_cr_west = cr - west
    diff_east_west = east - west
    
    # Percent differences (relative to comparison group)
    pct_cr_east = (diff_cr_east / east) * 100
    pct_cr_west = (diff_cr_west / west) * 100
    pct_east_west = (diff_east_west / west) * 100
    
    results.append({
        "Season": season,
        
        "CR - East (km)": diff_cr_east,
        "CR - East (%)": pct_cr_east,
        
        "CR - West (km)": diff_cr_west,
        "CR - West (%)": pct_cr_west,
        
        "East - West (km)": diff_east_west,
        "East - West (%)": pct_east_west
    })

diff_df = pd.DataFrame(results)

print("\nMedian Path Length Differences:")
display(diff_df.round(2))


Median Path Length Differences:


,Season,CR - East (km),CR - East (%),CR - West (km),CR - West (%),East - West (km),East - West (%)
0,DJF,30.96,590.18,28.74,384.91,-2.22,-29.74
1,MAM,7.89,150.77,8.77,201.85,0.89,20.37
2,JJA,5.79,324.32,5.01,194.37,-0.79,-30.63
3,SON,19.26,266.00,22.48,558.80,3.22,80.00


### Seasonal Differences in Median Tornado Path Length

To quantify how tornado path length varies by river interaction, median path length differences were computed between Crossed River, East Side, and West Side tornadoes for each season.

The largest differences are observed during **winter (DJF)**:

- **Crossed River vs East Side:** +30.96 km (**+590%**)  
- **Crossed River vs West Side:** +28.74 km (**+385%**)  

This represents the strongest enhancement in tornado path length across all seasons, indicating that **river-crossing tornadoes in winter are substantially longer-lived than nearby non-crossing tornadoes**.

Moderate but still substantial differences are observed during **spring (MAM)**:

- **Crossed River vs East Side:** +7.89 km (**+151%**)  
- **Crossed River vs West Side:** +8.77 km (**+202%**)  

These differences remain statistically significant and reinforce the strong springtime signal associated with river-crossing tornadoes.

During **summer (JJA)**:

- **Crossed River tornadoes** continue to exhibit longer median path lengths:
  - **+5.79 km (+324%) relative to East Side tornadoes**
  - **+5.01 km (+194%) relative to West Side tornadoes**

- East Side tornadoes exhibit slightly shorter median path lengths than West Side tornadoes:
  - **East vs West:** −0.79 km (**−31%**)

Although the absolute differences are smaller than during DJF, the relative percentage differences remain very large because summertime non-crossing tornadoes tend to have comparatively short path lengths.

In **fall (SON)**:

- **Crossed River vs East Side:** +19.26 km (**+266%**)  
- **Crossed River vs West Side:** +22.48 km (**+559%**)  

These large differences suggest that river-crossing tornadoes may again exhibit enhanced persistence during autumn, although the seasonal statistical signal is weaker overall due to increased variability and smaller sample sizes.

### Key Takeaway

Across all seasons, tornadoes that **cross the Mississippi River consistently exhibit longer median path lengths** than tornadoes confined to either side of the river buffer.

The effect is **most pronounced during winter (DJF)**, where Crossed River tornadoes exhibit dramatically larger median path lengths relative to both East Side and West Side tornadoes. Strong enhancements are also observed during spring and summer, while fall shows large but less statistically robust differences.

Overall, these findings provide strong evidence that **river interaction is associated with enhanced tornado persistence**, particularly during cool-season environments when synoptic forcing and low-level thermodynamic contrasts may be strongest.